### DISCUSSION TITLE
Better Offline Indicator

### DISCUSSION THREAD

**Discussion by carlosmbe** [specify/specify7#5891] [Feature Requests]
If you disconnect from WiFi while using Specify, it will throw a failed to fetch error. I think this is a bit overkill since we know that its failing to fetch due to internet connectivity, as a small icon will appear on the bottom right. Can we make have a pretty, "You're Offline" error instead. And can we capitalise the first letter of "Offline"/"Online" while at it.

<img width="1792" alt="Screenshot 2023-11-16 at 3 25 40 PM" src="https://github.com/specify/specify7/assets/53784701/58c6df25-be88-4838-a0a5-867e973c5c3e">

https://github.com/specify/specify7/assets/53784701/779aaf0a-2809-4a1a-8ed0-d3647e8b0ffb

![cover](https://github.com/specify/specify7/assets/53784701/6237fab1-e926-4349-a662-014d9bb7c0d8)
![maxresdefault](https://github.com/specify/specify7/assets/53784701/5f6178e6-0400-4dc6-a73f-69ed29dbe892)

**Comment by maxpatiiuk** [comment]
online indicator is not full-proof (see https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine)

the way we could do this is:
- Error code 421 (fake error thrown by Specify) appears in many different cases whenever it fails to make the network request (i.e error occurs before the request is even sent) - for example when previous request is canceled by the browser because the page begun reloading.
- so we should only display the offline message if browser detected we are offline AND response code is 421
